spi/topology/src/main/java/quarks/topology/spi/graph/ConnectorStream.java

```diff
@@ -163,6 +175,100 @@ protected Graph graph() {
     }
     
     @Override
+    public <J, U, K> TStream<J> join(Function<T, K> keyer,
+            TWindow<U, K> twindow, BiFunction<T, List<U>, J> joiner) {
+        
+        if(twindow instanceof TWindowImpl){
+            TStream<U> lastStream = twindow.feeder();
+            BiFunction<List<U>,K, Object> processor = Functions.synchronizedBiFunction((list, key) -> null);
+            Window<U, K, LinkedList<U>> window = Windows.lastNProcessOnInsert(((TWindowImpl<U, K>)twindow).getSize(), twindow.getKeyFunction());
+            Aggregate<U,Object,K> op = new Aggregate<U,Object,K>(window, processor);
+            lastStream.pipe(op);
+            return this.map((tuple) -> {
+                Partition<U, K, ? extends List<U>> part = window.getPartitions().get(keyer.apply(tuple));
+                if(part == null)
+                    return null;
+                J ret;
+                synchronized (part) {
+                    List<U> last = part.getContents();
+                    ret = joiner.apply(tuple, last);
+                }
+                return ret;
+            });
+        }
+        
+        else if (twindow instanceof TWindowTimeImpl){
+            TStream<U> lastStream = twindow.feeder();
+            BiFunction<List<U>,K, Object> processor = Functions.synchronizedBiFunction((list, key) -> null);
+            long time = ((TWindowTimeImpl<U, K>)(twindow)).getTime();
+            TimeUnit unit = ((TWindowTimeImpl<U, K>)(twindow)).getUnit();
+            Window<U, K, InsertionTimeList<U>> window =
+                    Windows.window(
+                            alwaysInsert(),
+                            scheduleEvictIfEmpty(time, unit),
+                            evictOlderWithProcess(time, unit),
+                            processOnInsert(),
+                            twindow.getKeyFunction(),
+                            insertionTimeList());
+            Aggregate<U,Object,K> op = new Aggregate<U,Object,K>(window, processor);
```

Comment:
Can some explanation be added, it's unclear why an Aggregate is used that is unconnected.

I guess i was expecting a new Join oplet to be created & used.